diff -Naur dkms-2.2.0.3.old/dkms dkms-2.2.0.3/dkms
--- dkms-2.2.0.3.old/dkms	2014-06-16 10:17:12.681812092 +0200
+++ dkms-2.2.0.3/dkms	2014-06-19 11:09:28.637884577 +0200
@@ -753,19 +753,19 @@
 {
     # $@ = files to process
     # do nothing if we have no obsoletes
-    [[ ${modules_conf_obsoletes[@]} ]] || return 0
+    ( IFS=; [[ "${modules_conf_obsoletes[*]}" ]] ) || return 0
     # generate sed args to remove obsolete modules
     local mod_diff
     for ((index=0; index < ${#dest_module_name[@]}; index++)); do
 	[[ ${modules_conf_obsoletes[$index]} ]] || continue
 	for obsolete_module in ${modules_conf_obsoletes[$index]//,/ }; do
 	    # for module.conf style syntax
-	    sa_mc_o[${#sa_mc_o[@]}]="-e"
-	    sa_mc_o[${#sa_mc_o[@]}]="s/\(alias ${modules_conf_alias_type[$index]}[0-9]*\) $obsolete_module$/\1 ${dest_module_name[$index]}/g"
+	    sa_mc_o[${#sa_mc_o[*]}]="-e"
+	    sa_mc_o[${#sa_mc_o[*]}]="s/\(alias ${modules_conf_alias_type[$index]}[0-9]*\) $obsolete_module$/\1 ${dest_module_name[$index]}/g"
 
 	    # for /etc/sysconfig/kernel style syntax
-	    sa_sck_o[${#sa_sck_o[@]}]="-e"
-	    sa_sck_o[${#sa_sck_o[@]}]="s/\(INITRD_MODULES.*\)$obsolete_module\b\(.*\)/\1${dest_module_name[$index]}\2/"
+	    sa_sck_o[${#sa_sck_o[*]}]="-e"
+	    sa_sck_o[${#sa_sck_o[*]}]="s/\(INITRD_MODULES.*\)$obsolete_module\b\(.*\)/\1${dest_module_name[$index]}\2/"
 	done
     done
 
@@ -773,9 +773,9 @@
     for file in "$@"; do
 	[[ $file && -w $file ]] || continue
 	if [[ $file = /etc/sysconfig/kernel ]]; then
-	    sed "${sa_sck_o[@]}" "$file" > "$temp_dir_name/${file##*/}.new"
+	    sed "${sa_sck_o[*]}" "$file" > "$temp_dir_name/${file##*/}.new"
 	else
-	    sed "${sa_mc_o[@]}" "$file" > "$temp_dir_name/${file##*/}.new"
+	    sed "${sa_mc_o[*]}" "$file" > "$temp_dir_name/${file##*/}.new"
 	fi
 	if ! mod_diff=$(diff -u "$temp_dir_name/${file##*/}.new" "$file"); then
 	    echo $"$file updated to replace obsoleted module references:"
